pkg/webui/account/containers/profile-settings-form/index.js

```diff
@@ -0,0 +1,292 @@
+    : getClosestProfilePictureBySize(value, 128)
+}
+
+const imageUriFromEmbedded = value =>
+  `data:${value.embedded.mime_type};base64,${value.embedded.data}`
+
+const ProfileEditForm = () => {
+  const dispatch = useDispatch()
+
+  const user = useSelector(selectUser)
+  const useGravatarConfig = useSelector(selectUseGravatarConfiguration)
+  const disableUploadConfig = useSelector(selectDisableUploadConfiguration)
+  const initialProfilePictureSource =
+    useGravatarConfig && isGravatarProfilePicture(user.profile_picture) ? 'gravatar' : 'upload'
+
+  const [profilePictureSource, setProfilePictureSource] = useState(initialProfilePictureSource)
+  const [error, setError] = useState()
+  const [emailAddress, setEmailAddress] = useState(user.primary_email_address)
+
+  const { debouncedFunction: debouncedSetEmail, cancel: debounceCancel } = debounce(
+    setEmailAddress,
+    500,
+  )
+
+  const validationContext = {
+    disableUploadConfig,
+    useGravatarConfig,
+    initialProfilePictureSource,
+  }
+
+  useEffect(() => () => debounceCancel(), [debounceCancel])
+
+  const handleEmailAddressChange = useCallback(
+    value => {
+      debouncedSetEmail(value)
+    },
+    [debouncedSetEmail],
+  )
+
+  const gravatarPreview = useMemo(
+    () =>
+      convertUriToProfilePicture(
+        `https://gravatar.com/avatar/${md5(emailAddress.toLowerCase())}?s=256&d=identicon`,
+      ),
+    [emailAddress],
+  )
+
+  const handleSubmit = useCallback(
+    async (values, { setSubmitting, resetForm }) => {
+      setError(undefined)
+      let patch = diff(user, validationSchema.cast(values, { context: validationContext }), [
+        '_profile_picture_source',
+      ])
+      if (Object.keys(patch).length === 0) {
+        patch = { ...values }
+        delete patch.profile_picture
+        delete patch._profile_picture_source
+      }
+      try {
+        const updatedUser = await dispatch(
+          attachPromise(updateUser({ id: user.ids.user_id, patch })),
+        )
+        resetForm({
+          values: {
+            ...user,
+            ...updatedUser,
+            _profile_picture_source: values._profile_picture_source,
+          },
+        })
+        toast({
+          title: sharedMessages.success,
+          message: m.successMessage,
+          type: toast.types.SUCCESS,
+        })
+      } catch (error) {
+        setError(error)
+        setSubmitting(false)
+      }
+    },
+    [dispatch, user, validationContext],
+  )
+
+  const handleDelete = useCallback(async () => {
+    try {
+      await dispatch(attachPromise(deleteUser)())
+
+      // The hard redirect will conclude the deletion by deleting the
+      // (now invalid) session cookie and redirecting back to the login screen.
+      // The `account-deleted` query will cause a success notification to be
+      // shown on the login screen.
+      const appRoot = selectApplicationRootPath()
+      window.location = `${appRoot}/login?account-deleted`
```

Comment:
Not really robust approach. Redirect with a state seems to be more appropriate here

Reply:
We cannot do that here because we need to reinitialize the AA with a refresh after the account was deleted and all sessions and access tokens were removed. This basically equals logging out, which we also do with a hard refresh, which to me is the safest way to reinitialize, as opposed to clearing all state manually.

Reply:
I see. My initial comment was not about using hard reload with `window.location`, but about the approach for displaying info notification via query parameter. I just want to emphasize that this shouldnt become the normal way to handle such cases.

Im fine with keeping this for now, but later for similar cases this should be implemented properly with more robust approach.